File: CommonBase/src/main/java/com/common/base/ability/IPresenter.java
package com.common.base.ability;

import com.common.base.BasePresenter;

/**
 * Presenter集成接口
 *
 * @author LiuFeng
 * @data 2021/10/15 15:49
 */
public interface IPresenter<P extends BasePresenter> {

    /**
     * 创建Presenter
     *
     * @return
     */
    P createPresenter();
}

File: CommonBase/src/main/java/com/common/base/ability/IPresenter.kt
package com.common.base.ability

import com.common.base.BasePresenter

/**
 * Presenter集成接口
 *
 * @author LiuFeng
 * @data 2021/10/15 15:49
 */
interface IPresenter<P : BasePresenter<out IBaseView>> {
    /**
     * 创建Presenter
     *
     * @return
     */
    fun createPresenter(): P
}